From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Thu, 3 Sep 2026 11:41:27 +0200
Subject: feat(debian.conf): prefer busybox

Include busybox on Debian/Ubuntu if available. This reduces the size of
the initrd significantly. Some data for Ubuntu 26.04 LTS "resolute"
using the current `ubuntu:rolling` container with a modified test 10.
The data is collected with `ls -l` and `3cpio --examine`.

## Without plymouth

An initrd without plymouth and without busybox uses 42.0 MB (42001866
bytes) when zstd compressed:

```
Start     End       Size      Compr.   Extracted
0 B       92.0 kB   92.0 kB   cpio     91.1 kB
92.0 kB   17.3 MB   17.2 MB   cpio     17.0 MB
17.3 MB   42.0 MB   24.7 MB   zstd     77.9 MB
```

The same initrd with busybox needs only 37.9 MB (37861705 bytes) when
zstd compressed:

```
Start     End       Size      Compr.   Extracted
0 B       92.0 kB   92.0 kB   cpio     91.1 kB
92.0 kB   17.3 MB   17.2 MB   cpio     17.0 MB
17.3 MB   37.9 MB   20.6 MB   zstd     65.4 MB
```

So 4.1 MB (9.9 %) disk space is saved and 12.5 MB (16.0 %) memory after
decompression is saved.

## With plymouth

An initrd with plymouth and without busybox uses 49.5 MB (49493709
bytes) when zstd compressed:

```
Start     End       Size      Compr.   Extracted
0 B       92.0 kB   92.0 kB   cpio     91.1 kB
92.0 kB   17.3 MB   17.2 MB   cpio     17.1 MB
17.3 MB   49.5 MB   32.2 MB   zstd     99.4 MB
```

The same initrd with busybox needs only 45.3 MB (45348147 bytes) when
zstd compressed:

```
Start     End       Size      Compr.   Extracted
0 B       92.0 kB   92.0 kB   cpio     91.1 kB
92.0 kB   17.3 MB   17.2 MB   cpio     17.1 MB
17.3 MB   45.3 MB   28.0 MB   zstd     86.9 MB
```

So 4.1 MB (8.4 %) disk space is saved and 12.5 MB (12.6 %) memory after
decompression is saved.

Bug-Ubuntu: https://bugs.launchpad.net/bugs/2150657
Origin: upstream, https://github.com/dracut-ng/dracut/commit/aa2229dc9e272029e7e98f95bec954d0a64fe697
---
 dracut.conf.d/debian/01-debian.conf | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dracut.conf.d/debian/01-debian.conf b/dracut.conf.d/debian/01-debian.conf
index 8402c1b..e01e2dc 100644
--- a/dracut.conf.d/debian/01-debian.conf
+++ b/dracut.conf.d/debian/01-debian.conf
@@ -1,3 +1,4 @@
 # Debian/Ubuntu specific Dracut configuration
 i18n_vars="/etc/locale.conf:LANG /etc/default/console-setup:FONT,FONT_MAP,CONSOLE_MAP-FONT_UNIMAP /etc/default/keyboard:XKBLAYOUT-KEYMAP,XKBVARIANT-EXT_KEYMAPS"
 initrdname="initrd.img-${kernel}"
+prefer_dracutmodules+=" busybox "
